STYLE.md

```diff
@@ -52,6 +52,8 @@
   exceptions such as `main()` functions. When code cannot be placed inside the
   Envoy namespace there should be a comment of the form `// NOLINT(namespace-envoy)` at
   the top of the file.
+* If a public method is only intended to be called from test code then it should
+  have a name that ends in ForTest() such as aMethodForTest().
```

Comment:
Thanks for adding this convention!

Is this also true for protected methods?  Would "If a method defined outside of the test directory is only intended..." be more clear or too wordy?